### PR TITLE
Yoga zed merge

### PR DESCRIPTION
**Pull request by Alex-Welsh**
Syncing latest stackhpc/yoga into Zed. Only real conflicts were in pulp.yml where formatting was changed to reduce repetition.

### CODE REVIEW

**Comment by markgoddard** [comment]
Weird pep8 reno failure

File: etc/kayobe/environments/aufn-ceph/globals.yml [hunk 1]
```diff
@@ -10,4 +10,7 @@ os_distribution: "{{ lookup('pipe', '. /etc/os-release && echo $ID') | trim }}"
 
 # OS release. Valid options are "8-stream" when os_distribution is "centos", or
 # "focal" when os_distribution is "ubuntu".
-#os_release:
+os_release: >-
+          {{ (lookup('pipe', '. /etc/os-release && echo $VERSION_CODENAME') | trim) if os_distribution == 'ubuntu' else
+             (lookup('pipe', '. /etc/os-release && echo $VERSION_ID') | trim | split('.') | first) if os_distribution == 'rocky' else
+             'stream-8' }}
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/globals.yml]: Not a good default any more.

File: etc/kayobe/environments/ci-aio/automated-setup.sh [hunk 2]
```diff
@@ -6,6 +6,11 @@ cat << EOF | sudo tee -a /etc/hosts
...
 EOF
 
+if [ sudo vgdisplay | grep -q lvm2 ]; then
+   sudo lvextend -L 4G /dev/rootvg/lv_home -r || true
+   sudo lvextend -L 4G /dev/rootvg/lv_tmp -r || true
+fi
+
 BASE_PATH=~
 KAYOBE_BRANCH=stackhpc/yoga
 KAYOBE_CONFIG_BRANCH=stackhpc/yoga
```
Comment (markgoddard) [inline etc/kayobe/environments/ci-aio/automated-setup.sh]: These will need to change.

File: etc/kayobe/environments/ci-aio/globals.yml [hunk 3]
```diff
@@ -51,7 +51,10 @@ os_distribution: "{{ lookup('pipe', '. /etc/os-release && echo $ID') | trim }}"
 
 # OS release. Valid options are "8-stream" when os_distribution is "centos", or
 # "focal" when os_distribution is "ubuntu".
-#os_release:
+os_release: >-
+          {{ (lookup('pipe', '. /etc/os-release && echo $VERSION_CODENAME') | trim) if os_distribution == 'ubuntu' else
+             (lookup('pipe', '. /etc/os-release && echo $VERSION_ID') | trim | split('.') | first) if os_distribution == 'rocky' else
+             'stream-8' }}
```
Comment (markgoddard) [inline etc/kayobe/environments/ci-aio/globals.yml]: As above

File: etc/kayobe/overcloud-dib.yml [hunk 4]
```diff
@@ -7,8 +7,8 @@
 # Whether to build host disk images with DIB directly instead of through
 # Bifrost. Setting it to true disables Bifrost image build and allows images to
 # be built with the `kayobe overcloud host image build` command. Default value
-# is true.
-#overcloud_dib_build_host_images:
+# is {{ os_distribution == 'rocky' }}. This will change in a future release.
```
Comment (markgoddard) [inline etc/kayobe/overcloud-dib.yml]: The default has changed

**Comment by Alex-Welsh** [comment]
> Weird pep8 reno failure

Spent a while looking at that last week. I think the issue is the git tagging is all off.
Unfortunately I based the `stackhpc/zed` branch on `stackhpc/yoga`, rather than upstream `stable/zed`. The automatic tagging from `naming-things-is-hard` therefore finds a common ancestor way back in yoga, and gets the tags wrong. 
I think the fix is to:

1.  branch stable/zed (tmp/zed)
2. merge in stackhpc/zed
3. delete stackhpc/zed
4. Rename tmp/zed ->stackhpc/zed

But that would need your magical admin powers.

I tried a few different merge/rebase style solutions but nothing quite kept all the changes properly. There might be a way but my git wrangling skills don't quite stretch that far

File: etc/kayobe/environments/aufn-ceph/globals.yml (hunk 1 above)
Comment (Alex-Welsh) [reply etc/kayobe/environments/aufn-ceph/globals.yml]: Fixed

File: etc/kayobe/environments/ci-aio/globals.yml (hunk 3 above)
Comment (Alex-Welsh) [reply etc/kayobe/environments/ci-aio/globals.yml]: Fixed

**Review by markgoddard — DISMISSED** [review]

File: etc/kayobe/environments/ci-aio/automated-setup.sh (hunk 2 above)
Comment (Alex-Welsh) [reply etc/kayobe/environments/ci-aio/automated-setup.sh]: Done

File: etc/kayobe/overcloud-dib.yml (hunk 4 above)
Comment (Alex-Welsh) [reply etc/kayobe/overcloud-dib.yml]: Done

**Comment by Alex-Welsh** [comment]
Closing in favour of https://github.com/stackhpc/stackhpc-kayobe-config/pull/620 due to my mistake rebasing this branch on stackhpc/zed

**Pull request closed without merging**